src/Backends/WaylandBackend.cpp

```diff
@@ -2002,24 +2002,32 @@ namespace gamescope
         // Prefer opaque for composition on the Wayland backend.
 
         uint32_t u8BitFormat = DRM_FORMAT_INVALID;
-        if ( SupportsFormat( DRM_FORMAT_XRGB8888 ) )
-            u8BitFormat = DRM_FORMAT_XRGB8888;
-        else if ( SupportsFormat( DRM_FORMAT_XBGR8888 ) )
-            u8BitFormat = DRM_FORMAT_XBGR8888;
-        else if ( SupportsFormat( DRM_FORMAT_ARGB8888 ) )
-            u8BitFormat = DRM_FORMAT_ARGB8888;
-        else if ( SupportsFormat( DRM_FORMAT_ABGR8888 ) )
-            u8BitFormat = DRM_FORMAT_ABGR8888;
+        for (int i = 0; i < 2 && u8BitFormat == DRM_FORMAT_INVALID; i++)
+        {
+            bool invalidOnly = i == 0;
+            if ( SupportsFormat( DRM_FORMAT_XRGB8888, invalidOnly ) )
```

Comment:
`SupportsFormat(format, false)` will always return true if `SupportsFormat(format, true)` returns true. In other words, I don't think we need to ever check `SupportsFormat(format, true)` here: checking whether the modifier list is non-empty should be enough.

Reply:
>SupportsFormat(format, false) will always return true if SupportsFormat(format, true) returns true.

That is so, however, if DRM_FORMAT_ARGB8888 supports the invalid modifier but DRM_FORMAT_XRGB8888 only supports explicit modifiers, then the old code and my patch will select DRM_FORMAT_ARGB8888. Whereas what you're suggesting would select DRM_FORMAT_XRGB8888.

Reply:
Why should the invalid modifier be preferred? It should be the other way around: explicit modifiers should be preferred when supported.

(Note, in practice formats will either all support implicit modifiers, or none.)

Reply:
>Why should the invalid modifier be preferred?

Why indeed. My goal for this PR was not to change the behavior in any case where gamescope already worked, only to enable new cases where gamescope would otherwise outright crash. So that the PR could get merged quickly.

Whoever authored SupportsFormat thought that checking for the invalid modifier explicitly instead of just checking if the list was not empty was important.

If you think it's fine, I can reduce this PR to a one-line patch that just changes how SupportsFormats works.

Reply:
I don't think the current behavior was entirely intentional, and since in practice it doesn't make a difference, I'd prefer simpler code and preferring explicit modifiers.

(I still think we should check for the invalid modifier in `vulkan_init_format()`, though.)

Reply:
This should be done.